### PR TITLE
 school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by ridabensalem**
In this milestone of the School library project we have added the following requirements : 

1. Create a class Person with the following:
2. Instance vars: @id, @name, and @age.
3. Constructor with name, age, and parent_permission as a parameter. 
4. Getters for @id, @name, and @age.
5. Setters for @name and @age.
6. Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
7. Public method can_use_services? that returns true if the person is of age or if they have permission from parents.

- Create class Student with the following:
1. Inherits from Person.
2. Constructor extends parent's constructor by adding @classroom and a parameter for it.
3. Method play_hooky that returns "¯\(ツ)/¯".
4. Create a class Teacher with the following:
5. Inherits from Person.
6. Constructor extends parent's constructor by adding @specialization and a parameter for it.
7. Override can_use_services? so it always returns true

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  def initialize(id, name = 'Unknown', age = 0, specialization:, parent_permission: true)
+    super(id, name, age, parent_permission: parent_permission)
+    @specialization = specialization
+  end
+
+  def can_use_services?
+    true
+  end
+end
```
Comment (mukhammadrizooff) [inline student.rb]: - [ ] [REQUIRED] Good job so far! There should be an individual `.rb`(`Person`, `Student`, and `Teacher`) file for each class 🙏🏼

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class Person
+  attr_accessor :id, :name, :age
+
```
Comment (mukhammadrizooff) [inline person.rb]: - [ ] [REQUIRED] Good job so far! The `:id` should only be able to be retrieved so only the Getter is defined, please replace the `:id` to `attr_reader` 🙏🏼 

```ruby
  attr_accessor :name, :age 
  attr_reader :id 
```

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  def initialize(id, name = 'Unknown', age = 0, parent_permission: true)
+    @id = id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```
Comment (mukhammadrizooff) [inline person.rb]: - [ ] [REQUIRED] Good job so far! Kindly remove `id` from all  `initialize()` and give a value as like below🙏🏼 

```ruby
def initialize(age, name = "Unknown", parent_permission = true)
    @id = Random.rand(1..1000)
    @age = age
    @name = name
    @parent_permission = parent_permission
  end
```

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @ridabensalem, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShadyShawkat — APPROVED** [review]
## Hi @ridabensalem, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**